rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -522,6 +522,7 @@ private ExprCfgNode getALastEvalNode(ExprCfgNode e) {
   result = e.(BreakExprCfgNode).getExpr() or
   result = e.(BlockExprCfgNode).getTailExpr() or
   result = e.(MatchExprCfgNode).getArmExpr(_) or
+  result = e.(MacroExprCfgNode).getMacroCall().(MacroCallCfgNode).getExpandedNode() or
```

Comment:
Should this rather be
```ql
result = e.(MacroExprCfgNode).getMacroCall() or
result = e.(MacroCallCfgNode).getExpandedNode() or
```
?

Reply:
Conceptually yes, but `MacroCall` is not an expression and doesn't have a node in the data-flow graph, so it won't work. To do that I think we'd have to change the type of `getALastEvalNode` and add a new kind of data-flow node for `MacroCall`.

ThaSo just adding a step over the `MacroCall` seems simpler and is also what we do for the other kind of nodes that have their expression nested further down.

We could add a method on `MacroExprCfgNode` to get the expanded node directly?

Reply:
Thanks, let's leave it as-is then.